dashboard/src/api/types.ts

```diff
@@ -0,0 +1,68 @@
+import * as geojson from 'geojson';
+
+export interface AuthToken {
+    token: string;
+}
+
+export interface CodeToken {
+    token: string;
+}
+
+//////////////////////////////////////////////////////////////////////
+// Primitive types
+
+type Url = string;
+type ParkingAreaId = string;
+type RegionId = string;
+
+//////////////////////////////////////////////////////////////////////
+// Helper interfaces
+
+interface PaginatedList {
+    count: number;
+    next: Url|null;
+    previous: Url|null;
+}
+
+// Type of the Features returned by the API
+interface ApiFeature<I extends string, P> extends
+geojson.Feature<geojson.MultiPolygon, P> {
+    type: 'Feature';
+    geometry: geojson.MultiPolygon;
+    id: I;
+    properties: P|null;
+}
+
+// Type of the Feature Collections returned by the API
+interface ApiFeatureCollection<I extends string, P> extends
+geojson.FeatureCollection<geojson.MultiPolygon, P> {
+    type: 'FeatureCollection';
+    features: Array<ApiFeature<I, P>>;
+}
+
+//////////////////////////////////////////////////////////////////////
+// Exported API interfaces
+
+export type Region = ApiFeature<RegionId, RegionProperties>;
+
+export interface RegionList extends
```

Comment:
This looks really unreadable, no idea if it just is just me not knowing TS as well or what but the last {} for instance looks really wierd.

Reply:
Are you referring to this? (Can't be sure, since the comment wasn't on the last line as I would expect it be on...)

```typescript
export interface RegionList extends
ApiFeatureCollection<RegionId, RegionProperties>,
PaginatedList {
}
```

It's declaring an interface which extends two other interfaces, but not adding any new properties. Don't know how to make it more readable...

Reply:
I refered to the entire export statement, but excecially to the last {}. Can't it be written as
```
export interface RegionList extends
ApiFeatureCollection<RegionId, RegionProperties>,
PaginatedList {}
```
for instance?